shpc/main/registry/remote.py

```diff
@@ -231,10 +205,48 @@ def iter_registry(self, filter_string=None):
             # Assemble a faux config with tags so we don't hit remote
             yield RemoteResult(uri, entry, load=False, config=entry["config"])
 
+    def get_container_url(self, module_name):
+        raise NotImplementedError
+
+    def get_raw_container_url(self, module_name):
+        raise NotImplementedError
+
 
 class GitHub(VersionControl):
-    provider_name = "github"
+    @classmethod
+    def matches(cls, source):
+        return urllib.parse.urlparse(source).hostname == "github.com"
+
+    @property
+    def library_url(self):
+        url_path = self.parsed_url.path.lstrip("/")
+        if url_path.endswith(".git"):
+            url_path = url_path[:-4]
+        owner, repo = url_path.split("/", 1)
+        return f"https://{owner}.github.io/{repo}/library.json"
+
+    def get_container_url(self, module_name):
+        return f"https://github.com/{self.parsed_url.path}/blob/{self.tag}/{module_name}/container.yaml"
+
+    def get_raw_container_url(self, module_name):
+        return f"https://raw.githubusercontent.com/{self.parsed_url.path}/{self.tag}/{module_name}/container.yaml"
 
 
 class GitLab(VersionControl):
-    provider_name = "gitlab"
+    @classmethod
+    def matches(cls, source):
+        return urllib.parse.urlparse(source).hostname == "gitlab.com"
+
+    @property
+    def library_url(self):
+        url_path = self.parsed_url.path.lstrip("/")
+        if url_path.endswith(".git"):
+            url_path = url_path[:-4]
+        owner, repo = url_path.split("/", 1)
+        return f"https://{owner}.gitlab.io/{repo}/library.json"
+
+    def get_container_url(self, module_name):
```

Comment:
These are different enough that I like how they are separate - thanks for doing this change I also like it better!